src/shared/utils/utils.tsx

```diff
@@ -108,3 +110,24 @@ export function* deepFindAllByKey<T = string>(
     }
   }
 }
+
+export const addBlastLinksToFreeText = (
+  texts: string[],
+  primaryAccession: string
+) =>
+  texts.map((text) => {
+    const splitText = text.split(/(\d+-\d+)/);
+    return splitText.map((splitItem, i) => {
+      if (i % 2 === 1) {
+        return (
+          <Link
+            to={getBLASTURL(primaryAccession, splitItem)}
```

Comment:
I would do something like

```suggestion
            to={getURLToJobWithData(JobTypes.BLAST, primaryAccession, splitItem)}
```

Or at least modify the existing getURLToJobWithData to directly handle splitItem if you don't want to split it by start and end